tests/functional/ldap/test_functional_revision.py

```diff
@@ -0,0 +1,214 @@
+from copy import deepcopy
+from datetime import datetime
+
+import pymongo
+import pytest
+
+from sarc.ldap.api import get_user, users_collection
+from sarc.ldap.revision import commit_matches_to_database, update_user
+
+statuses = ("disabled", "archived", "enabled")
+
+
+def make_user(status, **kwargs):
+    return {
+        "mila_ldap": {
+            "mila_email_username": "mail",
+            "mila_cluster_username": "posixUid",
+            "mila_cluster_uid": "uidNumber",
+            "mila_cluster_gid": "gidNumber",
+            "display_name": "displayName",
+            "supervisor": None,
+            "co_supervisor": None,
+            "status": status,
+        },
+        **kwargs,
+    }
+
+
+def user_dict(users):
+    return {u["mila_ldap"]["mila_email_username"]: u for u in users}
+
+
+def transitions():
+    all_transitions = []
+
+    for start_status in statuses:
+        for end_status in statuses:
+            all_transitions.append((start_status, end_status))
+    return all_transitions
+
+
+def dictset(dictionnary: dict, operation: dict):
+    result = deepcopy(dictionnary)
+
+    for path, value in operation.items():
+        current = result
+        frags = path.split(".")
+
+        for frag in frags[:-1]:
+            current = current.setdefault(frag, dict())
+
+        current[frags[-1]] = value
+
+    return result
+
+
+def insert_user_history(username, history):
+    users = users_collection().get_collection()
+
+    user_template = {
+        "name": username,
+        "mila": {
+            "username": username,
+            "email": "2",
+            "active": True,
+        },
+        "mila_ldap": {
+            "mila_email_username": username,
+            "supervisor": None,
+        },
+    }
+
+    for entry in history:
+        update_user(users, dictset(user_template, entry))
+
+
+def user_with_history(username):
+    insert_user_history(
+        username,
+        [
+            {"mila_ldap.supervisor": "1", "record_start": datetime(2001, 1, 1)},
+            {"mila_ldap.supervisor": "2", "record_start": datetime(2002, 1, 1)},
+            {"mila_ldap.supervisor": "3", "record_start": datetime(2003, 1, 1)},
+            {"mila_ldap.supervisor": "4", "record_start": datetime(2004, 1, 1)},
+        ],
+    )
+
+
+@pytest.mark.usefixtures("write_setup")
+def test_get_user_fetch_latest():
+
+    user_with_history("user1")
+
+    #
+    # Check latest
+    #
+    latest_user = get_user(mila_email_username="user1")
+    assert latest_user is not None, "Latest user was found"
```

Comment:
```suggestion
    assert latest_user is not None, "Latest user was not found"
```
Si le message est imprimé, c'est que `latest_user` est None et donc pas trouvé.

Reply:
Mais si tu lis le code, c'est la description de ce qu'on vérifie :p